### PR TITLE
Normalize users

### PR DESCRIPTION
**Pull request by dylanratcliffe**
This means that people can specify a list of user names that should be
in a role, they will be converted to UUIDs before being enforced

### CODE REVIEW

File: lib/puppet/provider/rbac_role/ruby.rb [hunk 1]
```diff
@@ -64,6 +64,22 @@ def destroy
     fail "The id parameter is read-only."
   end
 
+  def normalize_users(list)
+    users = nil
+    list.collect! do |item|
+      next item if item.to_i != 0
```
Comment (binford2k) [inline lib/puppet/provider/rbac_role/ruby.rb]: Is this intended to make it so you can invoke this on a mixed list of IDs and names?

Comment (dylanratcliffe) [reply lib/puppet/provider/rbac_role/ruby.rb]: Good catch, the users have a UUID and not a number, need to fix...

**Comment by dylanratcliffe** [comment]
@binford2k So the intention is that you can use UUIDs or names or a mix. But if you use names it will resolve them to UUIDs. This means that you can add users directly to roles without knowing their UUID (which you never do)

**Comment by dylanratcliffe** [comment]
Bump

**Comment by greglarkin** [comment]
@dylanratcliffe Is this module used in customer engagements?

**Comment by binford2k** [comment]
@greglarkin the vra_puppet_plugin_prep module uses it, Nick's pe_code_manager_webhook uses it, Example42 uses it in their psick system, etc. And a lot of customers use it in their profile or internal modules.

**Comment by greglarkin** [comment]
> @greglarkin the vra_puppet_plugin_prep module uses it, Nick's pe_code_manager_webhook uses it, Example42 uses it in their psick system, etc. And a lot of customers use it in their profile or internal modules.

Great, thanks for that info. I'm wondering if it makes sense to eventually get rid of the "pltraining-" portion of the module name if people are using it in production systems. Or maybe not an issue to deal with yet?

**Comment by dylanratcliffe** [comment]
@greglarkin Yeah loads of customer engagements, it's super useful

**Pull request merged**